release/pkg/generate_spec_test.go

```diff
@@ -0,0 +1,169 @@
+			cliMinVersion:       "v0.8.0",
+			cliMaxVersion:       "v0.8.0",
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.testName, func(t *testing.T) {
+			bundle := &anywherev1alpha1.Bundles{
+				Spec: anywherev1alpha1.BundlesSpec{
+					Number:        releaseConfig.BundleNumber,
+					CliMinVersion: tt.cliMinVersion,
+					CliMaxVersion: tt.cliMaxVersion,
+				},
+			}
+			bundle.APIVersion = "anywhere.eks.amazonaws.com/v1alpha1"
+			bundle.Kind = anywherev1alpha1.BundlesKind
+			bundle.CreationTimestamp = v1.Time{Time: releaseConfig.ReleaseDate}
+
+			homeDir, err := os.UserHomeDir()
+			if err != nil {
+				t.Fatalf("Error getting home directory: %v\n", err)
+			}
+
+			parentSourceDir := filepath.Join(homeDir, "eks-a-source")
+			err = os.RemoveAll(parentSourceDir)
+			if err != nil {
+				t.Fatalf("Error removing source directory: %v\n", err)
+			}
+
+			gitRoot, err := git.GetRepoRoot()
+			if err != nil {
+				t.Fatalf("Error getting top-level Git directory: %v\n", err)
+			}
+
+			releaseConfig.BuildRepoBranchName = tt.buildRepoBranchName
+			releaseConfig.CliRepoBranchName = tt.cliRepoBranchName
+			releaseConfig.ArtifactDir = filepath.Join(gitRoot, releaseFolder, artifactsDownloadLocation)
+
+			releaseVersion, err := releaseConfig.GetCurrentEksADevReleaseVersion(releaseConfig.ReleaseVersion)
+			if err != nil {
+				t.Fatalf("Error getting previous EKS-A dev release number: %v\n", err)
+			}
+
+			releaseConfig.ReleaseVersion = releaseVersion
+			releaseConfig.DevReleaseUriVersion = strings.ReplaceAll(releaseVersion, "+", "-")
+
+			err = os.RemoveAll(releaseConfig.ArtifactDir)
+			if err != nil {
+				t.Fatalf("Error removing local artifacts directory: %v\n", err)
+			}
+
+			err = releaseConfig.SetRepoHeads()
+			if err != nil {
+				t.Fatalf("Error getting heads of code repositories: %v\n", err)
+			}
+
+			bundleArtifactsTable, err := releaseConfig.GenerateBundleArtifactsTable()
+			if err != nil {
+				t.Fatalf("Error getting bundle artifacts data: %v\n", err)
+			}
+			releaseConfig.BundleArtifactsTable = bundleArtifactsTable
+
+			err = releaseConfig.PrepareBundleRelease()
+			if err != nil {
+				t.Fatalf("Error preparing bundle release: %v\n", err)
+			}
+
+			imageDigests, err := releaseConfig.GenerateImageDigestsTable(bundleArtifactsTable)
+			if err != nil {
+				t.Fatalf("Error generating image digests table: %+v\n", err)
+			}
+
+			err = releaseConfig.GenerateBundleSpec(bundle, imageDigests)
+			if err != nil {
+				t.Fatalf("Error generating bundles manifest: %+v\n", err)
+			}
+
+			bundleManifest, err := yaml.Marshal(bundle)
+			if err != nil {
+				t.Fatalf("Error marshaling bundles manifest: %+v\n", err)
+			}
+
+			expectedBundleManifestFile := filepath.Join(gitRoot, releaseFolder, testdataFolder, fmt.Sprintf("%s-bundle-release.yaml", tt.buildRepoBranchName))
+			generatedBundleManifestFile := filepath.Join(releaseConfig.ArtifactDir, fmt.Sprintf("%s-dry-run-bundle-release.yaml", tt.buildRepoBranchName))
```

Comment:
nit: I would pass this file names from the testcase

Reply:
I initially had it that way, but parametrized it since I wanted these filenames to be constructed using branch information from the test case.